### DISCUSSION TITLE
How to get `debug` or `release` profile name?

### DISCUSSION THREAD

**Discussion by jblachly** [rwf2/Rocket#1898] [Questions] [answered]
I'd like to have different behavior* in one of my routes depending on whether we are in development or production. I thought the most reasonable way would be to use a State extractor for the rocket config, ala:

```Rust
        .attach(AdHoc::config::<Config>())
```
and
```Rust
    let uri = match config.profile.as_str().as_str() {  // profile -> figment::UncasedStr -> str
        "debug" => outf_name,
        _ => String::from("s3://bucket/object"),
    };
```

However, the returned string is simply `default`, which I believe comes from Figment.

How can we access the /ROCKET/ profile, which is automatically set as `debug` in debug builds and `release` in release builds (https://rocket.rs/v0.5-rc/guide/configuration/#profiles) ?

**Comment by the10thWiz** [comment, marked as answer]
This is a bug, and it has been fixed on the master branch. Note that you don't need to attach the config as managed state, but you can actually just use `&Config` as a request guard to get the loaded config.
